### PR TITLE
feat: angular 5 support

### PR DESCRIPTION
**Pull request by crisbeto**
Bumps to Angular 5.0.0 and sorts out the various issues.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -25,26 +25,25 @@
...
+    "tsickle": "^0.24.x",
     "tslib": "^1.7.1",
     "zone.js": "^0.8.12"
   },
   "devDependencies": {
-    "@angular/compiler-cli": "~4.4.4",
-    "@angular/platform-browser-dynamic": "~4.4.4",
-    "@angular/platform-server": "~4.4.4",
-    "@angular/router": "~4.4.4",
-    "@angular/tsc-wrapped": "~4.4.4",
```
Comment (tbosch) [inline package.json]: As material used to use `tsc-wrapped` directly, but is now using `ngc`, you need to make sure to add `skipTemplateCodegen` to all of your `tsconfig.json` files, including the ones that inherit from a base `tsconfig.json` as ngc does not yet support inheritance. 

E.g. `src/material-moment-adapter/tsconfig-tests.json` is missing this setting.

Comment (crisbeto) [reply package.json]: Thanks, that sorted out the build error that I was running into. There's one more issue though: It seems like the compiler is dropping some of the imports (the ones that were purely used for type info). E.g. [this one](https://github.com/angular/material2/blob/master/src/cdk/overlay/overlay.ts#L21) is removed at compile time while the class' `ctorParameters` still reference it:

```js
Overlay.ctorParameters = function () { return [
    { type: index_1.ScrollStrategyOptions, },
    { type: overlay_container_1.OverlayContainer, },
    { type: core_1.ComponentFactoryResolver, },
    { type: overlay_position_builder_1.OverlayPositionBuilder, },
    { type: core_1.ApplicationRef, },
    { type: core_1.Injector, },
    { type: core_1.NgZone, },
]; };
```

**Comment by crisbeto** [comment]
Ready to go @jelbourn. Includes some of @DevVersion's changes from https://github.com/angular/material2/pull/8030.

**Comment by avril-verhaeghen** [comment]
Hi, any news about this PR? We want to migrate our project van MaterialJS to Angular Material2. For now we're still on 4.4.3 but are going to update to v5. Would be great if we could do both at the same time

**Comment by iTwangcj** [comment]
What's the problem with that? thanks

ERROR { Error: Uncaught (in promise): NAMESPACE_ERR: NAMESPACE_ERR (14): the operation is not allowed by Namespaces in XML
NAMESPACE_ERR: NAMESPACE_ERR (14): the operation is not allowed by Namespaces in XML
    at Object.exports.NamespaceError (/Users/wangchengjun/www/yooli/mock/angular-mock/node_modules/domino/lib/utils.js:28:45)
    at SVGSVGElement.setAttribute (/Users/wangchengjun/www/yooli/mock/angular-mock/node_modules/domino/lib/Element.js:422:50)
    at t.setAttribute (/Users/wangchengjun/www/yooli/mock/angular-mock/dist/server.js:128:10526)
    at t.e.setAttribute (/Users/wangchengjun/www/yooli/mock/angular-mock/dist/server.js:236:11870)
    at e.setAttribute (/Users/wangchengjun/www/yooli/mock/angular-mock/dist/server.js:278:24978)

**Comment by CaerusKaru** [comment]
@iTwangcj Are you on the latest version (nightly builds) of Material? If you are and are still having this issue, please open an issue documenting this (or we could reopen #7866).

**Comment by iTwangcj** [comment]
Thank you @CaerusKaru @willshowell 
I just found the problem two days ago, and I built a new angular5-demo(cli 1.5.0 automatically created).
You need to turn on the server rendering and app.component.html to add the following code:

<router-outlet></router-outlet>


<div class="fmape">
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 1440 810">
        <path fill="#efefee" d="M592.66 0c-15 64.092-30.7 125.285-46.598 183.777C634.056 325.56 748.348 550.932 819.642 809.5h419.672C1184.518 593.727 1083.124 290.064 902.637 0H592.66z"></path>
        <path fill="#f6f6f6" d="M545.962 183.777c-53.796 196.576-111.592 361.156-163.49 490.74 11.7 44.494 22.8 89.49 33.1 134.883h404.07c-71.294-258.468-185.586-483.84-273.68-625.623z"></path>
        <path fill="#f7f7f7" d="M153.89 0c74.094 180.678 161.088 417.448 228.483 674.517C449.67 506.337 527.063 279.465 592.56 0H153.89z"></path>
        <path fill="#fbfbfc" d="M153.89 0H0v809.5h415.57C345.477 500.938 240.884 211.874 153.89 0z"></path>
        <path fill="#ebebec" d="M1144.22 501.538c52.596-134.583 101.492-290.964 134.09-463.343 1.2-6.1 2.3-12.298 3.4-18.497 0-.2.1-.4.1-.6 1.1-6.3 2.3-12.7 3.4-19.098H902.536c105.293 169.28 183.688 343.158 241.684 501.638v-.1z"></path>
        <path fill="#e1e1e1" d="M1285.31 0c-2.2 12.798-4.5 25.597-6.9 38.195C1321.507 86.39 1379.603 158.98 1440 257.168V0h-154.69z"></path>
        <path fill="#e7e7e7" d="M1278.31,38.196C1245.81,209.874 1197.22,365.556 1144.82,499.838L1144.82,503.638C1185.82,615.924 1216.41,720.211 1239.11,809.6L1439.7,810L1439.7,256.768C1379.4,158.78 1321.41,86.288 1278.31,38.195L1278.31,38.196z"></path>
    </svg>
</div>

Run: NPM run start problem will appear. Thank you for offering solutions

**Comment by iTwangcj** [comment]
```html
app.component.html

<router-outlet></router-outlet>

<!-- The problem is here! -->
<div class="fmape">
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 1440 810">
        <path fill="#efefee" d="M592.66 0c-15 64.092-30.7 125.285-46.598 183.777C634.056 325.56 748.348 550.932 819.642 809.5h419.672C1184.518 593.727 1083.124 290.064 902.637 0H592.66z"></path>
        <path fill="#f6f6f6" d="M545.962 183.777c-53.796 196.576-111.592 361.156-163.49 490.74 11.7 44.494 22.8 89.49 33.1 134.883h404.07c-71.294-258.468-185.586-483.84-273.68-625.623z"></path>
        <path fill="#f7f7f7" d="M153.89 0c74.094 180.678 161.088 417.448 228.483 674.517C449.67 506.337 527.063 279.465 592.56 0H153.89z"></path>
        <path fill="#fbfbfc" d="M153.89 0H0v809.5h415.57C345.477 500.938 240.884 211.874 153.89 0z"></path>
        <path fill="#ebebec" d="M1144.22 501.538c52.596-134.583 101.492-290.964 134.09-463.343 1.2-6.1 2.3-12.298 3.4-18.497 0-.2.1-.4.1-.6 1.1-6.3 2.3-12.7 3.4-19.098H902.536c105.293 169.28 183.688 343.158 241.684 501.638v-.1z"></path>
        <path fill="#e1e1e1" d="M1285.31 0c-2.2 12.798-4.5 25.597-6.9 38.195C1321.507 86.39 1379.603 158.98 1440 257.168V0h-154.69z"></path>
        <path fill="#e7e7e7" d="M1278.31,38.196C1245.81,209.874 1197.22,365.556 1144.82,499.838L1144.82,503.638C1185.82,615.924 1216.41,720.211 1239.11,809.6L1439.7,810L1439.7,256.768C1379.4,158.78 1321.41,86.288 1278.31,38.195L1278.31,38.196z"></path>
    </svg>
</div>
```

**Comment by iTwangcj** [comment]
Test project address:
https://github.com/iTwangcj/angular5-seo-demo

**Pull request merged**